### ISSUE TITLE
Improve page name to folder name conversion

### ISSUE THREAD

**Issue by zzzzBov** [getgrav/grav-plugin-admin#1167]
Slugifying a name like "Can't touch this" produces "can-t-touch-this" because of how non-alphanumeric characters are translated.

Can apostrophes, and any other non-alphanumeric character that might appear mid-word be dropped rather than converted to hyphens?

**Comment by rhukster** [comment]
I think you are talking about the 'add page' modal?

<img width="739" alt="2017-07-25 at 6 38 pm" src="https://user-images.githubusercontent.com/1084697/28599545-8fba430a-7168-11e7-8c3e-a32c2fd982c1.png">

This is powered by a JS library.  I'll have to see if that can be configured.

**Comment by zzzzBov** [comment]
> I think you are talking about the 'add page' modal?

Yep

> This is powered by a JS library

I've done a little digging, it looks like [SpeakingURL][1] can take a custom hash to customize how specific characters are converted.

The custom [`$.slugify`][2] extension can be updated to pass the custom options by default:

```
$.slugify.options = {
    custom: {
      "'": ''
    },
    preSlug: null,
    postSlug: null,
    slugFunc: (input, opts) => getSlug(input, opts)
};
```

or [`add.js` can be updated to pass the custom options to slugify][3]:

```
let slug = $.slugify(elements.title.val(), {custom: {"'": ''}});
```

I'd make a pull request but then I'd also have to rebuild the JS, and I don't feel like jumping through all those hoops at the moment.

Hope that helps, and thanks for your time!

[1]: https://github.com/pid/speakingurl
[2]: https://github.com/getgrav/grav-plugin-admin/blob/develop/themes/grav/app/utils/jquery-utils.js#L46
[3]: https://github.com/getgrav/grav-plugin-admin/blob/develop/themes/grav/app/pages/page/add.js#L23

**Comment by rhukster** [comment]
Rebuilding JS is really not that bad, it's just a matter of doing an `npm install` to install all the bits, then a `gulp js` to build JS or `gulp watch-js` to watch and build when changes occur.

**Comment by w00fz** [comment]
Yes please go ahead with the PR, that's exactly the kind of implementation I would do for your proposed enhancement.
If you can't get the compilation going then just leave the PR allowing us to edit it and I'll recompile for you.

Thanks!

**Comment by w00fz** [comment]
This was merged in a PR (#1178). Thanks @zzzzBov
